community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/LuceneLabelScanStore.java

```diff
@@ -232,35 +257,51 @@ public void init() throws IOException
         monitor.init();
         directory = directoryFactory.open( directoryLocation );
 
-        try
-        {
-            // Try to open it, this will throw exception if index is corrupt.
-            // Opening it directly using the writer may hide corruption problems.
-            IndexReader.open( directory ).close();
-
-            writer = writerFactory.create( directory );
-        }
-        catch ( IndexNotFoundException e )
-        {
-            // No index present, create one
-            monitor.noIndex();
-            prepareRebuildOfIndex();
-            writer = writerFactory.create( directory );
-        }
-        catch ( LockObtainFailedException e )
+        if ( readOnly )
         {
-            monitor.lockedIndex( e );
-            throw e;
+            try
+            {
+                searcherManager = new SearcherManager( directory, new SearcherFactory() );
+            }
+            catch ( IOException e )
```

Comment:
Can we also catch `IndexNotFoundException` before this and report `monitor.noIndex()` and throw exception with more precise message?